lib/runtime/router.ts

```diff
@@ -251,6 +251,28 @@ export default class Router extends DenaliObject implements RouterDSL {
   }
 
   /**
+   * Returns the URL for a given action. You can supply a params object which
+   * will be used to fill in the dynamic segements of the action's route (if
+   * any).
+   */
+  urlFor(action: string | Action, data: Object): string | null {
+    if (typeof action === 'string') {
+      action = this.container.lookup(`action:${ action }`);
+    }
+    if (!action) {
+      return null;
+    }
+
+    let route: Route;
+    forEach(this.routes, (routes) => {
+      route = find(routes, { action: <Action> action });
```

Comment:
This type assertion shouldn't be required - TS is smart enough to know that the conditional at the start of the method is a type guard.

I think the clearer approach would be to type assert on line 260:

```
action = <Action>this.container.lookup(`action:${ action }`);
```